crates/core/component/dex/src/swap/proof.rs

```diff
@@ -133,8 +137,8 @@ impl ConstraintSynthesizer<Fq> for SwapCircuit {
         let balance_2 = BalanceVar::from_negative_value_var(swap_plaintext_var.delta_2_value());
         let balance_1_commit = balance_1.commit(transparent_blinding_var.clone())?;
         let balance_2_commit = balance_2.commit(transparent_blinding_var)?;
-        let transparent_balance_commitment = balance_1_commit + balance_2_commit;
-        let total_balance_commitment = transparent_balance_commitment + fee_commitment;
+        let opaque_commitment = fee_balance.commit(balance_blinding_var)?;
```

Comment:
the imbalance from `fee_balance` has already been captured and iterator consumed on constructing the `fee_commitment`.  calling `fee_balance.commit(balance_blinding_var)` with a new blinding factor $\widetilde{v_i}$ should only perform the blinding operation inside `commit`. 

$cv = [-v_1] G_1 + [-v_2] G_2 +[\widetilde{v_i}]G_{\widetilde{v_i}}$, where $\widetilde{v_i}$ is a private randomizer to make the commitment to the value balance hiding.